presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/CoordinatorConfigAPI.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller.api;
+
+import com.teradata.prestomanager.common.NodeSet;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.inject.Singleton;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+
+@Path("/coordinator/config")
+@Api(description = "the config API for coordinator")
+@Singleton
+public final class CoordinatorConfigAPI
+{
+    private static final Logger LOGGER = LogManager.getLogger(CoordinatorConfigAPI.class);
+
+    @GET
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Get available configuration files")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved configuration", response = String.class)})
+    public synchronized Response getConfig()
+    {
+        LOGGER.debug("GET /coordinator/config");
+        return newRequestBuilder(NodeSet.getCoordinator().resolve("/config")).buildGet().invoke();
+    }
+
+    @GET
+    @Path("/{file}")
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Get configuration by file")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved file", response = String.class),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response getConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file)
+    {
+        LOGGER.debug("GET /coordinator/config/{}", file);
+        return newRequestBuilder(NodeSet.getCoordinator().resolve("/config/" + file)).buildGet().invoke();
```

Comment:
This could also use UriBuilder: `UriBuilder.fromUri(...).segment("config", file).build()`.

(Though if you follow some of my suggestions regarding the coordinator on the other commit, it will need to change further.)